library/L2_HAL/displays/lcd/st7066u.hpp

```diff
@@ -167,56 +143,46 @@ class St7066u
   ///
   /// @param operation Operation transfer type.
   /// @param nibble    4-bit data to transfer.
-  void WriteNibble(WriteOperation operation, uint8_t nibble)
+  void WriteNibble(WriteOperation operation, uint8_t nibble) const
   {
-    kControlPins.e.Set(sjsu::Gpio::State::kHigh);
-    kControlPins.rs.Set(sjsu::Gpio::State(operation));
-    kControlPins.rw.Set(sjsu::Gpio::State::kLow);
-    // set nibble on 4-bit data bus
-    kControlPins.d7.Set(sjsu::Gpio::State((nibble >> 3) & 0x01));
-    kControlPins.d6.Set(sjsu::Gpio::State((nibble >> 2) & 0x01));
-    kControlPins.d5.Set(sjsu::Gpio::State((nibble >> 1) & 0x01));
-    kControlPins.d4.Set(sjsu::Gpio::State((nibble >> 0) & 0x01));
+    kEnablePin.Set(sjsu::Gpio::State::kHigh);
+    kRegisterSelectPin.Set(sjsu::Gpio::State(operation));
+    kReadWritePin.Set(sjsu::Gpio::State::kLow);
+
+    kDataBus.Write(nibble);
     sjsu::Delay(1ms);
```

Comment:
Does this have to be 1ms? or can it be shorter? Seems like a long time. Also, Delay can be used with nanoseconds precision now.